packages/handlersjs-http/lib/handlers/json-store-handler.ts

```diff
@@ -0,0 +1,80 @@
+import { from, Observable, of } from 'rxjs';
+import { TimedTypedKeyValueStore } from '@digita-ai/handlersjs-core';
+import { map, switchMap } from 'rxjs/operators';
+import { HttpHandler } from '../models/http-handler';
+import { HttpHandlerContext } from 'models/http-handler-context';
+import { HttpHandlerResponse } from 'models/http-handler-response';
+
+export class JsonStoreHandler<M> extends HttpHandler {
+
+  /**
+   * Creates a HTTP handler that returns the contents of the storage data as a stringified JSON response
+   *
+   * @param data the key for accessing the data in a given store
+   * @param store the store that contains the data
+   */
+  constructor(
+    private readonly data: keyof M,
+    private readonly store: TimedTypedKeyValueStore<M>,
+
+  ) {
+
+    super();
+
+  }
+
+  canHandle(context: HttpHandlerContext): Observable<boolean> {
+
+    return of(true);
+
+  }
+
+  /**
+   * Attempts to fetch the data from the storage
+   *
+   * @returns the stringified storage data as a HTTP response, or a Not Found HTTP response
+   */
+  private tryProvideData(): Observable<HttpHandlerResponse> {
+
+    return from(this.store.get(this.data)).pipe(map((data) => data ?
+      { body: JSON.stringify(data), headers: {}, status: 200 } : // OK
+      { body: '', headers: {}, status: 404 })); // not found
+
+  }
+
+  /**
+   * Handles an incoming http request to fetch the storage data
+   *
+   * @param context
+   * @returns an HTTP response
+   */
+  handle(context: HttpHandlerContext): Observable<HttpHandlerResponse> {
+
+    if (context.request.method === 'GET') {
+
+      const modifiedSince = context.request.headers['If-Modified-Since'];
```

Comment:
Not sure whether the headers are already transformed to a single case by the handlersjs server, but we should definitely take that into account here. If not, lower-case the headers to definitely find the right one if it is included in other casing.

Reply:
arthur verified it gets lowercased, when I asked him